homeassistant/components/slack/sensor.py

```diff
@@ -0,0 +1,52 @@
+"""Slack platform for select component."""
+
+from __future__ import annotations
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TIME_MINUTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.util.dt as dt_util
+
+from . import SlackEntity
+from .const import ATTR_SNOOZE, DATA_CLIENT, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Slack select."""
+    async_add_entities(
+        [
+            SlackSensorEntity(
+                hass.data[DOMAIN][entry.entry_id][DATA_CLIENT],
+                SensorEntityDescription(
+                    key="do_not_disturb_timer",
+                    name="Do Not Disturb Timer",
```

Comment:
Is this the time until a user is marked as DND?

Reply:
It's not clear what value this adds.

Reply:
No, Slack has DND set to pause notifications but an amount of time must be specified. This sensor indicates when DND expires.

Reply:
Ok, understood. Let's improve the name maybe? Maybe "Until" instead of "Timer" ?